### PR TITLE
Add initial version of miss_hit

### PR DESCRIPTION
**Pull request by mvanzulli**
Following: https://github.com/ONSAS/ONSAS/issues/677#issuecomment-2531204242 I added an initial local version of the formatter so we can start slowly disucssion conventions as well as set up details

### CODE REVIEW

File: miss_hit.cfg [hunk 1]
```diff
@@ -0,0 +1,44 @@
+# Style guide configuration for MISS_HIT
+# Reference: https://florianschanda.github.io/miss_hit/style_checker.html
+
+# Maximum allowed length for a single line of code (in characters)
+# Helps maintain code readability without unnecessary horizontal scrolling.
```
Comment (mvanzulli) [inline miss_hit.cfg:5]: Probably here we want to align on what is going to be the convention as in the current version is still erroring:

File: utils/Makefile [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+DIRECTORIES := ./../docs/ ./../test/
+# TODO: Add docs examples and src
+
+.PHONY: all
+all: check
+
+.PHONY: install
+install:
+	@echo "Installing dependencies..."
+	poetry install
```
Comment (mvanzulli) [inline utils/Makefile:16]: In general, I prefer poetry than pip as it is modern easy to use as poetry ensures consistent and reproducible builds with its `poetry.lock` file and simplifies dependency and project management in a single `pyproject.toml`. Ref: https://nanthony007.medium.com/stop-using-pip-use-poetry-instead-db7164f4fc72. 

Should be enough to install it https://python-poetry.org/docs/ and run `make install` but maybe something else is needed to make sure it creates a local `.venv` folder as regular pip with all deps.

File: miss_hit.cfg (hunk 1 above)
Comment (mvanzulli) [reply miss_hit.cfg:5]:
```
Checking code formatting in ./../docs/ ./../test/...
poetry run mh_style ./../docs/ ./../test/
In ../docs/src/bringONSASmFilesToONSASdocs.m, line 1
| % md function used to conver .m files in the ONSAS repo to .md files in this repo
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ style: No copyright notice found in docstring [copyright_notice]
In ../docs/src/bringONSASmFilesToONSASdocs.m, line 6
| function bringONSASmFilesToONSASdocs
|          ^^^^^^^^^^^^^^^^^^^^^^^^^^^ style: violates naming scheme for function [naming_functions]
In ../docs/src/m2md.m, line 1
| function m2md(fileIn, fileOut, includeCodeBoolean, iniLine)
|          ^^^^ style: violates naming scheme for function [naming_functions]
In ../docs/src/m2md.m, line 1
| function m2md(fileIn, fileOut, includeCodeBoolean, iniLine)
|          ^^^^ style: Could not find any copyright notice [copyright_notice]
In ../test/gaussIntegrationTest.m, line 1
| % ========================================================================================
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ style: No copyright notice found in docstring [copyright_notice]
In ../test/gaussIntegrationTest.m, line 4
| function resultBoolean = gaussIntegrationTest()
|                          ^^^^^^^^^^^^^^^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/gaussIntegrationTest.m, line 56
| function ys = test_fun_to_integrate(xinput)
|               ^^^^^^^^^^^^^^^^^^^^^ style: violates naming scheme for function [naming_functions]
../test/runTestProblems_local.m: style: violates naming scheme for scripts [naming_scripts]
In ../test/runTestProblems_moxunit_disp.m, line 1
| % function for testing ONSAS using moxunit
| ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ style: No copyright notice found in docstring [copyright_notice]
In ../test/runTestProblems_moxunit_disp.m, line 3
| function test_suite = runTestProblems_moxunit_disp
|                       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 16
| function test_1
|          ^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 20
| function test_2
|          ^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 24
| function test_3
|          ^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 28
| function test_4
|          ^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 32
| function test_5
|          ^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 36
| function test_6
|          ^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 40
| function test_7
|          ^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 44
| function test_8
|          ^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 48
| function test_9
|          ^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 52
| function test_10
|          ^^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 56
| function test_11
|          ^^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 60
| function test_12
|          ^^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 64
| function test_13
|          ^^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 68
| function test_14
|          ^^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 72
| function test_15
|          ^^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 76
| function test_16
|          ^^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 80
| function test_17
|          ^^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 84
| function test_18
|          ^^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 88
| function test_19
|          ^^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 92
| function test_20
|          ^^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 96
| function test_21
|          ^^^^^^^ style: violates naming scheme for function [naming_functions]
In ../test/runTestProblems_moxunit_disp.m, line 100
| function test_22
|          ^^^^^^^ style: violates naming scheme for function [naming_functions]
MISS_HIT Style Summary: 5 file(s) analysed, 32 style issue(s)
make: *** [Makefile:26: format_check] Error 1
```

File: utils/Makefile (hunk 2 above)
Comment (jorgepz) [reply utils/Makefile:16]: ok by me!

File: docs/src/m2md.m [hunk 3]
```diff
@@ -1,47 +1,47 @@
...
+      currentLine = fgetl(fidIn);
+    end
 
-  if length( currentLine )>=7 && strcmp( currentLine((end-6):end), '%hidden' )
-    % hidden line do not do anything
+    if length(currentLine) >= 7 && strcmp(currentLine((end - 6):end), '% hidden')
+      % hidden line do not do anything
 
-  elseif length( currentLine )>=3 && strcmp( currentLine(1:3), '%md' ) % not code
+    elseif length(currentLine) >= 3 && strcmp(currentLine(1:3), '% md') % not code
```
Comment (jorgepz) [inline docs/src/m2md.m:23]:
```suggestion
    elseif length(currentLine) >= 4 && strcmp(currentLine(1:4), '% md') % not code
```

Comment (jorgepz) [reply docs/src/m2md.m:23]: this is one of the lines I think should be edited to have the deploy working.

File: miss_hit.cfg (hunk 1 above)
Comment (jorgepz) [reply miss_hit.cfg:6]: Yes, however this could be solved in another PR? maybe this pr adds miss_hit and in another we define the format...

**Comment by jorgepz** [comment]
Thanks for the PR @mvanzulli ! This is a very nice improvement for the project.

**Review by jorgepz — CHANGES_REQUESTED** [review]
The deployment of the docs is not working since the length of the %md string changed

File: miss_hit.cfg (hunk 1 above)
Comment (mvanzulli) [reply miss_hit.cfg:6]: Makes sense, we can do that. Merge the machinery and then we can decide, I was thinking the automation of pre-commit can be added in a different PR after we clean all formats issues.

**Comment by mvanzulli** [comment]
> The deployment of the docs is not working since the length of the %md string changed

Good point I found that, proably we will need to support both or strip spaces while we do the migration. However I see http://www.onsas.org/previews/PR741/ is deployed correctly in this PR and why the tests are passing ?

File: miss_hit.cfg (hunk 1 above)
Comment (mvanzulli) [reply miss_hit.cfg:6]: I've tried to describe the plan https://github.com/ONSAS/ONSAS/issues/677#issue-2010341755, feel free to edit or modify as you consider the best to do a safe and sound auto formatting migration.

File: docs/src/m2md.m (hunk 3 above)
Comment (mvanzulli) [reply docs/src/m2md.m]: Got you, thanks for the MR. This one should be ready to go.

**Pull request merged**